### PR TITLE
fix: sql refactor & api results

### PR DESCRIPTION
**Pull request by stanislavec**
SQL issues were fixed & api responses enchanced to return at leat empty arrays to avoid iteration errors

### CODE REVIEW

File: plugins/time-saver-backend/src/database/TimeSaverDatabase.ts [hunk 1]
```diff
@@ -605,7 +614,7 @@ export class TimeSaverDatabase implements TimeSaverStore {
           this.db.raw(`DATE(created_at) as date`),
           this.db.raw('SUM(time_saved) as total_time_saved'),
         )
-        .groupBy('template_name', 'date');
+        .groupBy('team', 'date');
```
Comment (tduniec) [inline plugins/time-saver-backend/src/database/TimeSaverDatabase.ts]: here is the change with a logic, should by grouped by template_name.

Comment (stanislavec) [reply plugins/time-saver-backend/src/database/TimeSaverDatabase.ts]: Reverted

**Review by toma2395 — APPROVED** [review]
LGTM

**Review by tduniec — APPROVED** [review]
LGTM

**Pull request merged**